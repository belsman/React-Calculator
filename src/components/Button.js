import PropTypes from 'prop-types';

const Button = props => {
  const { buttonName, buttonClass, clickHandler } = props;
  return (
    <button type="button" onClick={() => clickHandler(buttonName)} className={buttonClass}>
      { buttonName }
    </button>
  );
};

Button.propTypes = {
  buttonName: PropTypes.string,
  buttonClass: PropTypes.string,
  clickHandler: PropTypes.func.isRequired,
};

Button.defaultProps = {
  buttonClass: 'button',
  buttonName: 'button',
};

export default Button;
