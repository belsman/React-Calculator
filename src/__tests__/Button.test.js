import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Button from '../components/Button';

describe('Button component test', () => {
  test("should renders '5' text", () => {
    render(<Button buttonName="5" clickHandler={() => ''} />);
    const text = screen.getByText('5');
    expect(text).toBeInTheDocument();
  });

  test('should render the expected output', () => {
    const component = renderer.create(
      <Button buttonName="5" clickHandler={() => ''} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
