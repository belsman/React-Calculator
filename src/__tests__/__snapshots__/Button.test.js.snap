// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test should render the expected output 1`] = `
<button
  className="button"
  onClick={[Function]}
  type="button"
>
  5
</button>
`;
